examples/cpp/benchmark/PublisherApp.cpp

```diff
@@ -0,0 +1,545 @@
+                        " Bytes) RECEIVED" << std::endl;
+                    sent = 0;
+                    if (elapsed.count() >= end_)
+                    {
+                        cv_.notify_one();
+                        return;
+                    }
+                    if (benchmark_big_.index() > count)
+                    {
+                        benchmark_big_.index(count);
+                    }
+                    else
+                    {
+                        benchmark_big_.index(benchmark_big_.index() + 1);
+                    }
+
+                    count = benchmark_big_.index() + 1;
+                    if ((RETCODE_OK == writer_->write(&benchmark_big_)) == true)
+                    {
+                        sent = 1;
+                        std::cout << "Sample with index: '" <<
+                            benchmark_big_.index() << "' (" << static_cast<int>(benchmark_big_.data().size()) <<
+                            " Bytes) SENT" << std::endl;
+                    }
+                }
+            }
+            break;
+
+        default:
+            throw std::runtime_error("Type invalid");
+    }
+}
+
+void PublisherApp::run()
+{
+    uint16_t prevCount = 0;
+    while (!is_stopped() && !publish())
+    {
+        // Wait for period
+        std::unique_lock<std::mutex> initial_lock(mutex_);
+        auto check = cv_.wait_for(initial_lock, std::chrono::milliseconds(1), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+    }
+    {
+        // Wait for period
+        std::unique_lock<std::mutex> wait_lock(mutex_);
+        auto check = cv_.wait_for(wait_lock, std::chrono::milliseconds(wait_), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+        count = 0;
+    }
+
+    auto actualTime = std::chrono::steady_clock::now();
+    auto elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(actualTime - startTime);
+    while (!is_stopped() && elapsed.count() < end_)
+    {
+        // Wait for period or stop event
+        std::unique_lock<std::mutex> periodic_lock(mutex_);
+        auto check = cv_.wait_for(periodic_lock, std::chrono::milliseconds(period_ms_), [&]()
+                        {
+                            return is_stopped();
+                        });
+        if (check)
+        {
+            return;
+        }
+        vSamples.push_back(static_cast<uint16_t>(count) - prevCount);
+        prevCount = static_cast<uint16_t>(count);
+        actualTime = std::chrono::steady_clock::now();
+        elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(actualTime - startTime);
+    }
+    std::cout << "RESULTS after " << elapsed.count() << " milliseconds:" << std::endl;
```

Comment:
As we are collecting some interesting data (i.e the number of messages exchanged for each period of time), it could be a good idea to add another result metric of how many bytes per second it has achieved (a kind of `throughtput` measurement)